pkg/pgmodel/metrics/database/metrics.go

```diff
@@ -45,6 +45,30 @@ type metricQueryWrap struct {
 	isHealthCheck bool
 }
 
+func compressionChunksQueries(countDelayed bool) string {
+	delayClause := "(m.delay_compression_until IS NULL OR m.delay_compression_until < now())"
+	if countDelayed {
+		delayClause = "(m.delay_compression_until IS NOT NULL AND m.delay_compression_until >= now())"
+	}
+
+	return `WITH chunk_candidates AS MATERIALIZED (
+		    SELECT chcons.dimension_slice_id, h.table_name, h.schema_name
+		    FROM _timescaledb_catalog.chunk_constraint chcons
+			     INNER JOIN _timescaledb_catalog.chunk c ON c.id = chcons.chunk_id
+			     INNER JOIN _timescaledb_catalog.hypertable h ON h.id = c.hypertable_id
+		    WHERE c.dropped IS FALSE
+		      AND h.compression_state = 1 -- compression_enabled = TRUE
+		      AND (c.status & 1) != 1) -- only check for uncompressed chunks
+		SELECT count(*)::BIGINT
```

Comment:
it would be more efficient to execute one query for both the delayed and not delayed case 
```suggestion
		SELECT count(*)::BIGINT FILTER(WHERE m.delay_compression_until IS NULL OR m.delay_compression_until < now()) as chunks_to_compress,
		SELECT count(*)::BIGINT FILTER(WHERE m.delay_compression_until IS NOT NULL AND m.delay_compression_until >= now()) as chunks_delayed
```

but I am not sure how hard it would be to build in multi-column/multi-metric infrastructure or if we already have it

Reply:
done